src/Parser.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace CSVParser;
+
+class Parser {
+
+    /**
+     * @var path the file path to the csv file to be parsed.
+     */
+    private $path;
+
+    /**
+     * @var headers The column headers for the CSV file that give context to the column values.
+     */
+    private $headers;
+
+    /**
+     * @var delimiter the delimiter value that separates column values for this CSV file.
+     */
+    private $delimiter;
+
+    function __construct ($path, $delimiter = ',', $headers = []) 
+    {
+        /*
+         * Check that the path does exist or throws an exception
+         * Checks that the delimiter is a , | or /t
+         * Fills headers, if it's empty grabs first line from CSV instead
+         */
+
+        if(!file_exists($path)) 
+            throw new \Exception("{$path} is not a valid file.");
+        $this->path = $path; //fill path
+
+        if(!in_array($delimiter, [',', '|', '\t']))  //verify delimiter
+            throw new Exception("A delimiter of {$delimiter} is not supported.");    
+        $this->delimiter = $delimiter; //fill delimiter
+
+        
+        if(empty($headers)) 
+        {
+            $this->setHeaders($path, $headers);
+        } else {
+            $this->headers = $headers;
+        }
+    }
+
+    public function getHeaders()
+    {
+        return $this->headers;
+    }
+
+    public function setHeaders($path, $headers = [])

```

Comment:
Imagine someone using the library like this
```php
$parser = new CSVParser($file);
$parser->setHeaders($file, ['col_1', 'col_2', 'col_3'])
echo $parser->getHeaders() . PHP_EOL;
```
Given the following file what are the proper header values?
```csv
ID, Name, Title, Years
1, Derek, Senior, 7
2, Arthur, Associate, 2 
```

Also what happened to the header values I passed to `setHeaders()`?
The setHeaders funtion may work standalone but used along with the rest of the library will cause weird issues. It's better to just use the constructor as I defined rather than giving another way to set headers. Also what is the use case for wanting to set header values halfway through the parsing a CSV file? the set headers function would allow for me to read 0 to N lines and then set headers to something totally random.

Reply:
I suppose it helps to understand that a valid CSV file only has one set of Header Values.

Reply:
It has been removed.  It was put in because prior instruction was that setters and getters are always best practice but I can understand it might create issues.